packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -36,7 +36,8 @@ Future<void> buildMacOS({
   required bool verboseLogging,
   SizeAnalyzer? sizeAnalyzer,
 }) async {
-  if (!flutterProject.macos.xcodeWorkspace.existsSync()) {
+  final Directory? xcodeWorkspace = flutterProject.macos.xcodeWorkspace;
+  if (xcodeWorkspace == null || !xcodeWorkspace.existsSync()) {
```

Comment:
I think the `existsSync` is not necessary now?
```suggestion
  if (xcodeWorkspace == null) {
```

Reply:
Fixed in https://github.com/flutter/flutter/pull/114099/commits/16e9dead270152baf07ac2ea35e8c9df3b49ef4f.